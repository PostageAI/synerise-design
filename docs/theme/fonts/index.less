@font-face {
  font-family: 'Graphik LCG Web';
  src: url('Graphik-Light-Cy-Gr-Web.eot');
  src: url('Graphik-Light-Cy-Gr-Web.eot?#iefix') format('embedded-opentype'),
    url('Graphik-Light-Cy-Gr-Web.woff2') format('woff2'), url('Graphik-Light-Cy-Gr-Web.woff') format('woff');
  font-weight: 300;
  font-style: normal;
  font-stretch: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@font-face {
  font-family: 'Graphik LCG Web';
  src: url('Graphik-Regular-Cy-Gr-Web.eot');
  src: url('Graphik-Regular-Cy-Gr-Web.eot?#iefix') format('embedded-opentype'),
    url('Graphik-Regular-Cy-Gr-Web.woff2') format('woff2'), url('Graphik-Regular-Cy-Gr-Web.woff') format('woff');
  font-weight: 400;
  font-style: normal;
  font-stretch: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@font-face {
  font-family: 'Graphik LCG Web';
  src: url('Graphik-Medium-Cy-Gr-Web.eot');
  src: url('Graphik-Medium-Cy-Gr-Web.eot?#iefix') format('embedded-opentype'),
    url('Graphik-Medium-Cy-Gr-Web.woff2') format('woff2'), url('Graphik-Medium-Cy-Gr-Web.woff') format('woff');
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@font-face {
  font-family: 'Graphik LCG Web';
  src: url('Graphik-Semibold-Cy-Gr-Web.eot');
  src: url('Graphik-Semibold-Cy-Gr-Web.eot?#iefix') format('embedded-opentype'),
    url('Graphik-Semibold-Cy-Gr-Web.woff2') format('woff2'), url('Graphik-Semibold-Cy-Gr-Web.woff') format('woff');
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@font-face {
  font-family: 'Graphik LCG Web';
  src: url('Graphik-Bold-Cy-Gr-Web.eot');
  src: url('Graphik-Bold-Cy-Gr-Web.eot?#iefix') format('embedded-opentype'),
    url('Graphik-Bold-Cy-Gr-Web.woff2') format('woff2'), url('Graphik-Bold-Cy-Gr-Web.woff') format('woff');
  font-weight: 700;
  font-style: normal;
  font-stretch: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
