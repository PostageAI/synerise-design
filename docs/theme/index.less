@import './fonts/index.less';

body * {
  font-family: 'Graphik LCG Web', sans-serif !important;
}

.ds-package-info {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 24px;
  border: 1px solid #ccc;
  margin-bottom: 16px;
}

#rsg-root {
  main {
    font-size: 16px;
    line-height: 28px;
    font-width: 400;
    color: #141414;

    a,
    button {
      box-sizing: border-box;
      color: rgb(14, 105, 255);
      cursor: pointer;
      display: inline-block;
      font-size: 16px;
      font-weight: 500;
      border-bottom: 0;
    }
    h1 {
      font-size: 48px;
      color: #141414;
      font-weight: 300;
      line-height: 1.2em;
    }
    h2 {
      font-size: 30px;
      font-stretch: 100%;
      font-style: normal;
      font-weight: 500;
      color: #141414;
      letter-spacing: normal;
      line-height: 36px;
    }
    h3 {
      font-size: 18px;
      font-weight: 600;
      line-height: 21.996px;
      color: #141414;
    }
    ul {
      li {
        ul {
          list-style-type: '- ';
        }
      }
    }

    pre {
      background-color: #f9f9f9;
    }
  }
  .rsg--sidebar-4 {
    background-color: #f9f9f9;
    header {
      border-bottom: 0;
      h1 {
        font-size: 30px;
      }
    }
  }
}
