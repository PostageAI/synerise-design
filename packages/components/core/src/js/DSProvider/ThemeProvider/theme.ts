import vars from './variables';

export type ThemeProps = {
  variables: { [key: string]: string };
  palette: { [key: string]: string };
  variable: (name: string) => string | null;
};

export const theme: ThemeProps = {
  variables: vars.variables,
  palette: vars.colors,
  variable: function(name: string): string | null {
    return name ? this.variables[name.slice(1)] : null;
  },
};
