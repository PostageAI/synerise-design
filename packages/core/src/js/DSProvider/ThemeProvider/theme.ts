import variables from '../../../style/variables';

export type ThemeProps = {
  variables: { [key: string]: string };
  palette: { [key: string]: string };
  variable: (name: string) => string | null;
};

export const theme: ThemeProps = {
  variables: variables,
  palette: variables,
  variable: function(name: string): string | null {
    return name ? this.variables[name.slice(1)] : null;
  },
};
