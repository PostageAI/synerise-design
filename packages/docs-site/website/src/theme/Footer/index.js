/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';

function Footer() {
  const [openedLinks, setOpenedLinks] = React.useState(false);

  const toggleLinks = React.useCallback(() => {
    setOpenedLinks(!openedLinks);
  }, [openedLinks]);
  return (
    <footer className="c-main-footer">
      <section className="c-footer-links l-section">
        <div className="c-main-footer__columns">
          <div className="c-main-footer__first-column">
            <img className="c-main-nav__logo-image c-dynamic-image" alt="Synerise" src="https://strapi.synerise.com/uploads/7cce7cae97314cee865f977b9baf13e5.svg" />
            <ul className="c-social-list">
              <li>
                <a href="https://facebook.com/synerise" aria-label="Facebook" rel="nofollow">
                  <i className="c-dynamic-image c-dynamic-icon  icon-facebook-fill-m c-dynamic-icon__icon"></i>
                </a>
              </li>
              <li>
                <a href="https://www.instagram.com/synerise/" aria-label="Instagram" rel="nofollow">
                  <i className="c-dynamic-image c-dynamic-icon  icon-instagram-fill-m c-dynamic-icon__icon"></i>
                </a>
              </li>
              <li>
                <a href="https://www.linkedin.com/company/synerise/" aria-label="LinkedIn" rel="nofollow">
                  <i className="c-dynamic-image c-dynamic-icon   icon-linkedin-fill-m c-dynamic-icon__icon"></i>
                </a>
              </li>
              <li>
                <a href="https://twitter.com/synerise" aria-label="Twitter" rel="nofollow">
                  <i className="c-dynamic-image c-dynamic-icon  icon-twitter-fill-m c-dynamic-icon__icon"></i>
                </a>
              </li>
            </ul>
          </div>
            <div className={`c-links-box ${openedLinks ? 'expanded' : ''}`}>
              <span className="c-links-box__title" role="button"  onClick={() => toggleLinks()}>
                Colloid Design System
                <svg className="arrow-icon" xmlns="http://www.w3.org/2000/svg" width="16" height="16"
                     viewBox="0 0 24 24" fill="none" stroke="#0E69FF" stroke-width="2" stroke-linecap="round"
                     stroke-linejoin="round"><polyline points="6 9 12 15 18 9"></polyline></svg>
              </span>
              <ul className="c-links-box__submenu submenu-group-0">
                <li>
                  <a className="c-links-box__lnk" href="/about">About</a>
                </li>
                <li>
                  <a className="c-links-box__lnk" href="/develop">Develop</a>
                </li>
                <li>
                  <a className="c-links-box__lnk" href="/docs/palette">Guidelines</a>
                </li>
                <li>
                  <a className="c-links-box__lnk" href="/storybook-static/">Components</a>
                </li>
              </ul>
            </div>
        </div>
      </section>
    </footer>
  );
}

export default Footer;
