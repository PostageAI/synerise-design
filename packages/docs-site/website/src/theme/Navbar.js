import React from 'react';

const Navbar = (props) => {
  const [opened, setOpened] = React.useState(false);

  const toggleMenu = React.useCallback(() => {
    setOpened(!opened);
  }, [opened]);
  console.log(props);
  return (
    <header className="c-main-header">
      <nav className={`c-main-header__nav l-section ${opened ? 'expanded' : ''}`}>
        <a className="c-main-header__logo" title="Home" href="/">
          <img src="/images/logo-synerise.svg " alt="Synerise" />
        </a>
        <button onClick={() => toggleMenu()} type="button" className={`c-main-header__btn ${opened ? 'active' : ''}`} aria-label="Toggle menu">
          <div className="btn"><span></span><span></span><span></span></div>
        </button>
        <ul className="c-main-nav">
          <li className="hover-action-subitem c-main-nav__item">
            <a className="c-main-nav__item__link" title="Case studies" href="/about">
              <span className="c-main-nav__submenu__item__title">Colloid DS</span>
            </a>
          </li>
          <li className="hover-action-subitem c-main-nav__item">
            <a className="c-main-nav__item__link" title="Case studies" href="/docs/palette">
              <span className="c-main-nav__submenu__item__title">Guidelines</span>
            </a>
          </li>
          <li className="hover-action-subitem c-main-nav__item">
            <a className="c-main-nav__item__link" title="Case studies" href="/storybook-static/">
              <span className="c-main-nav__submenu__item__title">Storybook</span>
            </a>
          </li>
          <li className="hover-action-subitem c-main-nav__item c-main-nav__item--end">
            <a className="c-read-more c-main-nav__item__link" href="https://app.synerise.com/spa/login" target="_self">
              <span>Log in</span>
              <i className="c-read-more__icon icon-arrow-right-m" aria-hidden="false"></i>
            </a>
          </li>
        </ul>
      </nav>
    </header>
  )
};

export default Navbar;
