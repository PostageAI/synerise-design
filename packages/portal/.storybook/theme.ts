import { create } from '@storybook/theming';

export default create({
  base: 'light',
  fontBase: "'Graphik LCG Web', sans-serif",
  brandTitle: 'Synerise Design System',
  brandUrl: 'https://synerise.com',
  brandImage: 'https://strapi.synerise.com/uploads/0525bd360f03436f947a2d0fe501ad68.svg',
});
