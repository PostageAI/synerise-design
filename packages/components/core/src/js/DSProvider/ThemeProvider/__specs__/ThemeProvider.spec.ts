import theme from '../theme';

describe('ThemeProvier', () => {
  it('should match snapshot', () => {
    // ASSERT
    expect(theme).toMatchSnapshot();
  });
});
