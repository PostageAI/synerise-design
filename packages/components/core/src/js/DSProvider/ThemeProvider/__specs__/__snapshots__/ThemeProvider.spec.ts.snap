// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvier should match snapshot 1`] = `
Object {
  "breakpoints": Array [
    "768px",
    "960px",
    "1280px",
  ],
  "palette": undefined,
  "space": Array [
    0,
    8,
    12,
    16,
    24,
    32,
    48,
    64,
  ],
  "variable": [Function],
  "variables": undefined,
}
`;
